pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -963,14 +964,25 @@ impl DeltaLayerInner {
                 .blobs_at
                 .as_slice()
                 .iter()
-                .map(|(_, blob_meta)| format!("{}@{}", blob_meta.key, blob_meta.lsn))
+                .filter_map(|(_, blob_meta)| {
+                    if blob_meta.key.is_rel_dir_key() || blob_meta.key == DBDIR_KEY {
+                        // The size of values for these keys is unbounded and can
+                        // grow very large in pathological cases.
+                        None
+                    } else {
+                        Some(format!("{}@{}", blob_meta.key, blob_meta.lsn))
+                    }
+                })
                 .join(", ");
```

Comment:
semi-nit: this will create N string allocations in addition to extending one multiple times (probably not doubling in size).

fold with a single string would allow allocating those small strings..

if this was a single `Display` impl the tracing would probably run it against a large enough (at least on second attempt) thread local string which would be clear append-only.

Reply:
If this `Display` thing is not done, should we at least add a warning to the max keys vectored read can read (the constant) that in the worst case they will all be output to a warning string?

Reply:
It's not obvious, but if the read is oversized, then it can only contain one key (due to how the planner works).